### PR TITLE
Preserve outgoing headers

### PR DESCRIPTION
**Pull request by walles**
Before this change, outgoing headers were ignored, leading to my REST requests failing.

With this change in place, outgoing headers are now (mostly) preserved, and my REST calls succeed.

### CODE REVIEW

File: bootstrap.js [hunk 1]
```diff
@@ -9,25 +9,31 @@ const ResponseBuilder = require('./app/ResponseBuilder');
...
+        delete headers.host;
+
+        // This prevents "Received response with content-encoding: gzip, but failed to decode it"
+        delete headers['accept-encoding'];
+
         request({
             uri: requestedUrl,
             resolveWithFullResponse: true,
-            headers: {
-                'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/48.0.2564.116 Safari/537.36'
```
Comment (imjacobclark) [inline bootstrap.js]: Could we ensure this user agent is still passed through on the headers object, looks at present that this has been removed?

File: bootstrap.js [hunk 2]
```diff
@@ -9,25 +9,31 @@ const ResponseBuilder = require('./app/ResponseBuilder');
...
+
         if(requestedUrl == ''){
             res.send(index);
             return;
         }
 
+        let headers = Object.assign({}, req.headers);
+
+        // This prevents "Error [ERR_TLS_CERT_ALTNAME_INVALID]: Hostname/IP does not match certificate's altnames: Host: localhost"
+        delete headers.host;
```
Comment (imjacobclark) [inline bootstrap.js]: Could you write some unit tests for this functionality please?

File: bootstrap.js (hunk 1 above)
Comment (walles) [reply bootstrap.js]: If the target system depends on the user agent being set properly, having the proxy overwriting it with something else will just break things.

To me this behavior was really unexpected; maybe document the rationale for this?

File: bootstrap.js (hunk 2 above)
Comment (walles) [reply bootstrap.js]: That would make a lot of sense.

**Comment by walles** [comment]
Unfortunately I won't put more time into this; even with my changes I never managed to solve my original problem (quite possibly not because of the proxy, there were many moving parts).

Will comment on your reviews, then closing this.

**Pull request closed without merging**